app/controllers/inspect/graph/graphs_controller.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Inspect
+  module Graph
+    class GraphsController < ApplicationController
+      skip_after_action :verify_policy_scoped
+      skip_before_action :authenticate_user!
+      if Rails.env.production?
+        before_action :authenticate_basic_always, only: :show
+      end
+
+      layout "full"
+
+      def show
+        @primary_type = safe_get_primary_type
+        if @primary_type.nil?
+          render plain:
+                   "You don't have permission to view object type: #{params[:object_type].to_s.downcase.singularize}",
+                 status: :bad_request and return
+        end
+        @primary_id = params[:object_id].to_i
```

Comment:
Don't need to use the `to_i` here, it even stands out a little bit.

```suggestion
        @primary_id = params[:object_id]
```